go/store/nbs/journal_writer.go

```diff
@@ -454,6 +510,21 @@ func (wr *journalWriter) writeCompressedChunk(cc CompressedChunk) error {
 	return nil
 }
 
+// sendIndexRecord launches a goroutine that will enqueue
+// a record to the indexCh channel.
+func (wr *journalWriter) sendIndexRecord(r interface{}) {
+	if wr.indexWriter != nil {
+		wr.sendEg.Go(func() error {
+			select {
+			case wr.indexCh <- r:
+			}
+			return nil
+		})
```

Comment:
Sending these async with unbounded concurrency seems a step too far...what's the point of not enqueuing it onto the buffered channel directly?

Reply:
blocking on buffered channel is better? i wasn't really sure, the cleanup that way is definitely easier